pkgs/objective_c/lib/src/ns_dictionary.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'converter.dart';
+import 'internal.dart';
+import 'ns_enumerator.dart';
+import 'objective_c_bindings_generated.dart';
+
+// Ideally we'd mixin UnmodifiableMapBase, but it's an ordinary class. So
```

Comment:
There is a `UnmodifiableMapMixin` in `package:collection` that I also use in my wip PR of extension types for jnigen which you could use (I'm using the list version actually)